src/common/utils/gather-tokens.util.ts

```diff
@@ -26,7 +36,7 @@ export function gatherProfileTokens(
         .filter((role) => role !== "@everyone")
         .join(", ") || "",
     "user.messages": userDB.messages_sent,
-    "user.voice_time": convertMilisToString(userDB.voice_time),
+    "user.voice_time": getDuration((userDB.voice_time + userVoiceTime) / 1000),
```

Comment:
And also I don't think that we need to do that. We should count voice time when user also executed /profile command. So assume that we will count time on command use - voice_time from db will be always up to date.

Reply:
> And also I don't think that we need to do that. We should count voice time when user also executed /profile command. So assume that we will count time on command use - voice_time from db will be always up to date.

Are you talking about the userVoiceTime? Did I correctly understand that you want to update the db voice_time on use of "/profile"? If so, that's an awful move. Firstly, it just extra work, because eventually we still need to update voice_time then user leaves voice. Secondly, that just makes the "/profile" (execute to be precise) non-pure, meaning that it changes voice_time value, while should only send profile of user. 

If I did understand you incorrectly, please clarify your sentence.

Reply:
I don't know wdym by "extra work". You just create function that will be used when user leave VC and when user execute /profile command. It's that simple. (I done that in the past and everything worked as charm)

And I don't see any problems in "precise" time when user look up his profile. You don't need to show them their seconds when they have days for example in voice time.

If user have months in voice time: display month and day
If user have only days in voice time: display days and hours
If user have only some hours that can't be converted to days: display hours and minutes.
etc.

Reply:
You're literary just creating an extra step by sending data to db for no reason. I'm not talking about that it "won't" work, I'm talking about that it is stupid and non-pure in function way. 

We have, literary, no reason to even send the request to database. Why even care about updating current voice_time, for it to be rewritten again when user leaves the voice? For the more readable code? Don't make me chuckle. 

To simplify, it would work like you have data on client side, but you send it to server side, just to get it back again on client side. Stupid and non-logical.

Reply:
I re-check how exactly you calc current voice time in "gatherTokens" func, and now I see why you debating this hard on that topic. I didn't saw that you literally pulling user from cache and just add this time to his current and return new value. So it's always up to date. My bad.

Reply:
Yeah, no worries.